src/main/java/jenkins/branch/NamedExceptionsBranchPropertyStrategy.java

```diff
@@ -101,12 +101,19 @@ public NamedExceptionsBranchPropertyStrategy(@CheckForNull BranchProperty[] defa
     @NonNull
     @Override
     public List<BranchProperty> getPropertiesFor(SCMHead head) {
+        List<BranchProperty> properties = new ArrayList<>();
+
         for (Named named : namedExceptions) {
             if (named.isMatch(head)) {
-                return new ArrayList<>(named.getProps());
+                properties.addAll(named.getProps());
```

Comment:
This looks unrelated to the main purpose of the PR, and it seems like it could be a breaking change because branch jobs for existing projects could end up with different properties after the update. Why are the changes to this method needed?

Reply:
This changes is required to this functionality, I explain this in [previous comment](https://github.com/jenkinsci/branch-api-plugin/pull/160#issuecomment-520864423). In short if you define a named exception for branch master with property A and other named expection for master and feature/* branches with other property B the property B will be ignored. The strategy return a list of named but for how is written it always returns one.
Just an example of our configuration in production
![immagine](https://user-images.githubusercontent.com/4160180/80829393-53ed7a80-8be7-11ea-9539-79e6a3a8e2dc.png)

At the time of writing these test cases passed, unit test added to october fails. i will try to have a look for failing test cases

Reply:
Ok, but it still seems like that is totally separate from "Add BranchProperty support to OrganizationFolder", this change is "Allow more than one named exception to match a single branch".

"Add BranchProperty support to OrganizationFolder" is something that does not affect existing users, it is a new feature.

"Allow more than one named exception to match each individual branch" is a change in existing behavior for multibranch projects (it sounds like you consider it a bug fix), and in the worst case it could break existing users. Without this change, you can duplicate the property for the `master, support/*` exclusion in the `master` exclusion and remove `master` from the `master, support/*` exclusion to get things to work the way you want.

Reply:
as workaround you can duplicate the definition to each branch you would the property. But also keep them aligned for each expcetion. I think there is not reason to make an interfaces to return a list of named branch and than returns always one. Also since is not written an explict limitation that only the first named exception is taked than from the point of view of end user this looks a bug. Do you agree?

Reply:
>  (it sounds like you consider it a bug fix), and in the worst case it could break existing users

since does not works except for the first matching exception there are no users that have named exception for the master (simply because actually does not work). This mean that could be considered a not breaking changes.

Reply:
> from the point of view of end user this looks a bug. Do you agree?

I'm not sure, but my point is mostly that I think it is totally distinct from "Add BranchProperty support to OrganizationFolder", so I would open a separate PR with just that change so it can be discussed separately.

> I think there is not reason to make an interfaces to return a list of named branch and than returns always one

I'm not sure what you mean here. The class has a list of exceptions, but there is no documentation on whether more than one exception can match a single branch, so the fact that the existing code returns when it finds a matching exception rather than checking all of them seems like reasonable behavior to me.

> since does not works except for the first matching exception there are no users that have named exception for the master (simply because actually does not work)

The problem is that someone might have an exception for `master`, and then a later exception that would also match `master` (perhaps unintentionally, or accidentally), and so with this change the properties that `master` would get would change. Maybe that is a very minor risk and so we do not care, I don't know, but I think it should be discussed in a separate PR.

Reply:
Separate PR avaiable #193